src/firstboot/firstboot.c

```diff
@@ -675,13 +681,22 @@ static int prompt_hostname(int rfd) {
 
         for (;;) {
                 _cleanup_free_ char *h = NULL;
+                _cleanup_free_ char *fallback = NULL;
 
-                r = ask_string(&h, "%s Please enter hostname for new system (empty to skip): ", special_glyph(SPECIAL_GLYPH_TRIANGULAR_BULLET));
+                fallback = get_default_hostname();
```

Comment:
lacks oom check.

but i really don't like this approach. we shouldn't write out redundant config files. i.e. if the compiled-in default is x, and this is a setting where we already set x anyway if we find no config, then let's *not* write out a config that sets it. The less config we write, the btter.

This matters actaully, so that on upgrades these settings follow our defaults, and do not remain "sticky" to whatever was current on first install. i.e. we want that /etc/ ideally contains only *user* selection, and not automatic choices – to the level this is possible.

Reply:
See my comment on the new config value for more detailed.

TLDR: All the other prompts are also written when prompted and selecting the default.